cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,160 @@
+particular course or globally. We can create a new Waffle namespace called
+"course_apps", and the course app ID can be the name for the
+``CourseWaffleFlag`` for each plugin by convention.
+
+To enumerate the list of available apps, we will list the installed plugins,
+check which plugins are available for the current course, and call each plugin
+to get its metadata. Each plugin will support checking if it is available for a
+particular course. If the plugin marks itself as unavailable, it won't be
+listed in the API response.
+
+Only legacy apps will support the ``legacy_link``, it will not be a publicised
+part of the API so new/external plugin apps shouldn't use it. This link should
+only be provided for course apps that don't have a UI in the course authoring
+MFE. If a partial UI exists, the MFE settings view can always link back to the
+old studio view from there.
+
+The following is a sample class with the pattern for such plugins:
+
+.. code-block :: python
+
+    WIKI_ENABLED = CourseWaffleFlag(COURSE_APPS_WAFFLE_NAMESPACE, 'wiki', __name__)
+
+    class CourseApp:
+        app_id: str = 'wiki'
+        name: str = 'Wiki'
+        description: str = 'A short description of what the Wiki does.'
+        # Specify if this app is enabled by default. If the app is made available for a course
+        # should it also automatically be considered enabled.
+        default_enabled: bool = False
+
+        IS_AVAILABLE = WIKI_ENABLED
+
+        # This method will not be in the sample/base class, but will be added to
+        # existing course apps.
+        if LEGACY_APP:
+            @classmethod
+            def legacy_link(cls, course_key):
+                return f'some/link/to/{course_key}'
+
+        @classmethod
+        def is_enabled(cls, course_key):
+            # Some logic to check if the app is enabled for this course
+            # This will not vary from user-to-user in studio.
+            return True
+
+        @classmethod
+        def set_enabled(cls, course_key, user, enabled):
+            # Some logic to enable the app for this course.
+            # The user here isn't passed on for permission checking, but just
+            # for logging/auditing.
+            return enabled
+
+        @classmethod
+        def get_allowed_operations(cls, course_key, user):
+            # This should return a dictionary with at least the `enable` and `configure` keys.
+            return {
+                'enable': can_user_enable(course_key, user),
+                'configure': can_user_configure(course_key, user),
+            }
+
+
+For each existing course app, we can create such a class and have these class
+methods call back to the existing code for the same.
+
+Consequences
+============
+
+- A new course apps API that consistently uses only a single mechanism for
+  discovering course apps, determining their availability and enabling/disabling
+  them.
+- We still leave the more complex considerations of configuration to each
```

Comment:
What are some configurations that we would like each app to handle themselves? Can we explicitly list them here within the current scope?

Reply:
This will be different for each app. For example for teams it will be configuring a set of teams, for discussions it will be discussions settings etc. 

We wouldn't want to consolidate everything under the same API.

Reply:
Yes right - The API is simply encapsulating all of the apps for listing and toggling and is specific to pages and resources view. 👍 

thanks for updating that.